lib/liquid.lua

```diff
@@ -2673,12 +2684,14 @@ function Template:parse( text , parser_context)
     instance.interpreter = Interpreter:new(instance.parser)
     return instance
 end
-function Template:render( context, filterset, resourcelimit )
+function Template:render( context, filterset, resourcelimit, filesystem )
     -- body
     local t_interpretercontext = context or InterpreterContext:new({})
     local t_filterset = filterset or FilterSet:new()
     local t_resourcelimit = resourcelimit or ResourceLimit:new()
-    return self.interpreter:interpret( t_interpretercontext, t_filterset, t_resourcelimit )
+    local t_filesystem = filesystem or FileSystem:new()
```

Comment:
I don't get why they are initialised here when they also have defaults in the `Interpreter:interpret`.